src/main/java/net/snowflake/ingest/streaming/internal/RowBufferStats.java

```diff
@@ -38,18 +41,29 @@ class RowBufferStats {
   /** Display name is required for the registration endpoint */
   private final String columnDisplayName;
 
+  private final boolean enableDistinctValue;
+  private Set<Object> distinctValues;
+
   /** Creates empty stats */
   RowBufferStats(
-      String columnDisplayName, String collationDefinitionString, int ordinal, Integer fieldId) {
+      String columnDisplayName,
+      String collationDefinitionString,
+      int ordinal,
+      Integer fieldId,
+      boolean enableDistinctValue) {
     this.columnDisplayName = columnDisplayName;
     this.collationDefinitionString = collationDefinitionString;
     this.ordinal = ordinal;
     this.fieldId = fieldId;
+    this.enableDistinctValue = enableDistinctValue;
+    if (enableDistinctValue) {
+      this.distinctValues = new HashSet<>();
+    }
     reset();
   }
 
   RowBufferStats(String columnDisplayName) {
```

Comment:
this ctor is only called by test code. lets add isEnableDistinctValue to the signature, and make all the test classes run with both true and false.

Reply:
Added to signature and create parameterized tests.